herald.py

```diff
@@ -55,6 +82,43 @@ async def runHerald(ctx, args):
             await ctx.send("You do not have a chosen audio, run /herald link")
             return
 
+    elif len(args) == 2 and args[0] == 'start':
+        # If there are two arguments and the first is 'start' then the user is
+        # trying to update the starting time of their audio
+
+        if ctx.author.id in heraldDict.keys():
+            # User exists
+            start_time = float(args[1])
+
+            # start_time cannot be greater than the audio length
+            if start_time < heraldDict[ctx.author.id].audioLength:
+
+                heraldDict[ctx.author.id].startTime = start_time
+                audio = AudioSegment.from_file(heraldDict[ctx.author.id].mp3Link)
+
+                # Create a new audio that starts at the desired time
+                edited_audio = audio[start_time*millisecond_conversion:]
+
+                edited_audio.export(f'herald/{ctx.author.id}_edited.mp3', format='mp3')
+                heraldDict[ctx.author.id].editedMp3Link = f'herald/{ctx.author.id}_edited.mp3'
+
+                # Need to adjust duration if the adjusted audio makes the duration extend past the audio length
+                if (heraldDict[ctx.author.id].audioLength - start_time) < heraldDict[ctx.author.id].duration:
+                    heraldDict[ctx.author.id].duration = heraldDict[ctx.author.id].audioLength - start_time
+
+                await ctx.send("Start time updated")
+
+            else:
+                # The start time is greater than the length of the audio
+                await ctx.send("The start time requested is beyond the end of the audio")
+                return
+
+        
+        else:
+            # User does NOT exist so they need to first pick their audio
+            await ctx.send("You do not have a chosen audio, run /herald link")
+            return
+
```

Comment:
Should we add an elif for len(args) == 2 and args[0] is something other than 'duration' or 'start'? As implemented, the bot will say, "A link is needed, run /herald link" which will seem wrong to a user who has a song selected but tries to use /herald again with a different argument (e.g., volume).

Reply:
I will add an elif for just len(args) == 2